### PR TITLE
[NEW] Video conf message block

### PR DESCRIPTION
**Pull request by GleidsonDaniel**
## Proposed changes

Adds support for new video conf block.

## Issue(s)	

Close: #4628
Close: #4680
## How to test or reproduce

Test entering and exiting a video conf using the new block.

## Screenshots
Screen Shot 2022-10-18 at 16 29 28

Screen Shot 2022-10-18 at 16 29 38

Screen Shot 2022-10-18 at 16 29 42

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Jira [VC-21]

[VC-21]: https://rocketchat.atlassian.net/browse/VC-21?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/containers/UIKit/VideoConferenceBlock/components/styles.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import { StyleSheet } from 'react-native';
+
+import { useTheme } from '../../../../theme';
+import sharedStyles from '../../../../views/Styles';
+
+export default function useStyle() {
```
Comment (diegolmello) [inline app/containers/UIKit/VideoConferenceBlock/components/styles.ts:6]: Pretty cool :)

File: app/containers/UIKit/VideoConferenceBlock/components/styles.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	const { colors } = useTheme();
+	return StyleSheet.create({
+		container: { height: 108, flex: 1, borderWidth: 1, borderRadius: 2, marginTop: 8, borderColor: colors.conferenceCallBorder },
```
Comment (diegolmello) [inline app/containers/UIKit/VideoConferenceBlock/components/styles.ts:9]: Recently we increased most of the borderRadius to 4.
We can confirm with design team, but I think Figma might be outdated and we need to use 4 here already and on some other places below

**Review by diegolmello — CHANGES_REQUESTED** [review]
Task's description:
```
Should not be possible to:
- Edit message
- Quote
- Pin
- Star
- Reply in DM?
```

It looks like we didn't implement that.

File: app/containers/CustomIcon/mappedIcons.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 export const mappedIcons = {
-	'lamp-bulb': 59812,
```
[the remark this answers is not in the document]
Comment (GleidsonDaniel) [reply app/containers/CustomIcon/mappedIcons.js:2]: Man, I didn't remove anything, probably it wasn't updated so the drive file with that icon.
The only thing I did was get the drive file and add the icon I wanted.

[the remark this answers is not in the document]
Comment (diegolmello) [reply app/containers/CustomIcon/mappedIcons.js:2]: Can you add lamp-bulb then?

**Review by diegolmello — APPROVED** [review]

**Comment by ankar84** [comment]
@GleidsonDaniel @diegolmello 
Testing Rocket.Chat Experimental version 4.33.0.38311 (with that PR merged) on 4.8.6 server and have a feedback.
1. Enter second Jitsi call in different chat lead to black screen, so https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4628 not fixed by this PR unfortunately
2. Ghost participant is remain on mobile when you leave and rejoin same call in same chat (that ghost users not persist on web version of Jitsi call, so it is only bug of Jitsi SDK on mobile)

**Comment by ankar84** [comment]
Tested same mobile version on 5.3.2 server version and have same issue.
Important moment - second call in different chat works fine until someone come to that call and after that screen become black.
Actually current call become real call only after 2 or more participants join it, so black screen appears on real call start.

**Comment by S3xyGod** [comment]
I got the same error with @ankar84. When I initiate a call from device 1 to device 2 or web to device 2. Device 2 joins, it works. Then I ended the call from all devices and made a new call to device 2. When device 2 pressed join only black screen on device 2 but on device 1 still see device 1 already participation. Pressing the back button on device 2 will return to the previous screen, but on device 1, device 2 is still visible in the call and cannot join another call. Only when force closing the application or restarting the device will this stop and the above error will repeat with the 2nd call
Screenshot_1669609936

**Pull request merged**